### PR TITLE
Docker, fixes, database, and more

### PR DESCRIPTION
**Pull request by dongle-the-gadget**
# Changes
- Add Docker support
- Add experimental MongoDB support (`DB_HOST`)
  - Currently doesn't do stuff
  - Uses MongoDB Node.js driver instead of mongoose
- Fixes
  - Fixed bugs that caused registering commands and plugins to fail due to incorrect folder searching (should search from current folder instead of `src`)
  - Fixed bugs related to constants (merged from #6)
  - Fixed a bug related to button (syntax error)
  - Ignore .d.ts files
  - Fixed `ping` returning a negative roundtrip value
  - Fixed restart command (thanks to @lost-in-action and @KojiOdyssey)
- Other misc changes
  - Add `DEV_GUILD` for development
  - Exclude bots from member count

### CODE REVIEW

File: src/plugins/commander/commands/about.ts [hunk 1]
```diff
@@ -102,11 +102,13 @@ export default class AboutCommand extends Command {
     });
   }
 
-  private executePing(interaction: CommandInteraction): Promise<void> {
-    return interaction.reply({
-        content: `Websocket heartbeat: ${this.client.ws.ping}ms
-Roundtrip latency: ${Date.now() - interaction.createdTimestamp}ms`,
-    });
+  async executePing(interaction: CommandInteraction) {
+    var originalReply = await interaction.reply({
```
Comment (lavgup) [inline src/plugins/commander/commands/about.ts]:
```suggestion
    const originalReply = await interaction.reply({ 
```

Comment (dongle-the-gadget) [reply src/plugins/commander/commands/about.ts]: 👍

**Comment by dongle-the-gadget** [comment]
I have a habit for using var for everything 😆

**Comment by lavgup** [comment]
You could add the [prefer-const](https://eslint.org/docs/rules/prefer-const) ESLint rule to the `.eslintrc.json` file, seems like a few people are used to using `var` over `const`/`let` :)

**Comment by dongle-the-gadget** [comment]
Considering the fact that I use C#, this could be why.

**Comment by dongle-the-gadget** [comment]
For some dumb reason `this.client.application.owner` always return null in my bot.

**Pull request closed without merging**